File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long numSum = 0;
        int i = 0;
        while(i<nums.size()){
            
            if(nums[i]!=0) i++;

            else{
                long long count = 0;
                while(i<nums.size() && nums[i]==0){
                    count++;
                    i++;
                }
                numSum+=((count*(count+1))/2);
            }
        
        }

        return numSum;
    }
};

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long totalCount = 0;
        int count0 = 0;

        for(int i=0; i<nums.length; i++){
            if(nums[i]==0){
                count0++;
                totalCount+=count0;
            }else count0 = 0;
        }

        return totalCount;
    }
}